client/my-sites/guided-transfer/host-credentials-page/index.jsx

```diff
@@ -29,16 +40,34 @@ class HostCredentialsPage extends Component {
 		this.setFieldValue( fieldName, e.target.value );
 	}
 
+	redirectToCart = () => {
+		upgradesActions.addItem( cartItems.guidedTransferItem() );
+		page.redirect( `/checkout/${ this.props.siteSlug }` );
```

Comment:
I think this is probably fine, but as an alternative, you could consider using [`lib/route`'s `addSiteFragment`](https://github.com/Automattic/wp-calypso/blob/dcf83067030e472fdc2e60355e7ebf4b8ab8528e/client/lib/route/path.js#L50-L64).